FirebaseStorageUITests/FUIImageViewCategoryTests.m

```diff
@@ -0,0 +1,188 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+@import XCTest;
+@import FirebaseStorageUI;
+
+#import "UIImageView+FirebaseStorage.h"
+
+@class MockDownloadTask, MockStorageReference, MockCache;
+
+@interface MockDownloadTask : NSObject <FUIDownloadTask>
+@property (nonatomic, readwrite, getter=isCancelled) BOOL cancelled;
+@end
+
+@implementation MockDownloadTask
+
+- (void)cancel {
+  self.cancelled = YES;
+}
+
+@end
+
+
+@interface MockCache : NSObject <FUIImageCache>
+@property (nonatomic, readonly) NSMutableDictionary *cached;
+@end
+
+@implementation MockCache
+
+- (instancetype)init {
+  self = [super init];
+  if (self != nil) {
+    _cached = [NSMutableDictionary dictionary];
+  }
+  return self;
+}
+
+- (void)storeImage:(UIImage *)image forKey:(NSString *)key {
+  self.cached[key] = image;
+}
+
+- (UIImage *)imageFromDiskCacheForKey:(NSString *)key {
+  return self.cached[key];
+}
+
+- (UIImage *)imageFromMemoryCacheForKey:(NSString *)key {
+  return self.cached[key];
+}
+
+@end
+
+
+@interface MockStorageReference : NSObject <FUIStorageReference>
+@property (nonatomic, readwrite) NSString *fullPath;
+@end
+
+@implementation MockStorageReference
```

Comment:
I don't doubt that mocking makes the code less verbose, but it increases coupling/complexity for the sake of less boilerplate. Here you seem to be arguing that you'd rather have complexity than verbosity.